checks.d/mesos_slave.py

```diff
@@ -0,0 +1,186 @@
+        return self._get_json(url + '/state.json', timeout)
+
+    def _get_stats(self, url, timeout):
+        if self.version >= [0, 22, 0]:
+            endpoint = '/metrics/snapshot'
+        else:
+            endpoint = '/stats.json'
+        return self._get_json(url + endpoint, timeout)
+
+    def _get_constant_attributes(self, url, timeout):
+        state_metrics = None
+        if self.cluster_name is None:
+            state_metrics = self._get_state(url, timeout)
+            if state_metrics is not None:
+                self.version = map(int, state_metrics['version'].split('.'))
+                master_state = self._get_state('http://' + state_metrics['master_hostname'] + ':5050', timeout)
+                if master_state is not None:
+                    self.cluster_name = master_state['cluster']
+
+        return state_metrics
+
+    def check(self, instance):
+        if 'url' not in instance:
+            raise Exception('Mesos instance missing "url" value.')
+
+        url = instance['url']
+        instance_tags = instance.get('tags', [])
+        tasks = instance.get('tasks', [])
+        default_timeout = self.init_config.get('default_timeout', 5)
+        timeout = float(instance.get('timeout', default_timeout))
+
+        state_metrics = self._get_constant_attributes(url, timeout)
+        tags = None
+
+        if state_metrics is None:
+            state_metrics = self._get_state(url, timeout)
+        if state_metrics:
+            tags = [
+                'mesos_cluster:{0}'.format(self.cluster_name),
+                'mesos_pid:{0}'.format(state_metrics['pid']),
+                'mesos_node:slave'
+            ]
+            tags += instance_tags
+
+            for task in tasks:
+                for framework in state_metrics['frameworks']:
+                    for executor in framework['executors']:
+                        for t in executor['tasks']:
+                            if task.lower() in t['name'].lower() and t['slave_id'] == state_metrics['id']:
+                                task_tags = ['task_name:' + t['name']] + tags
+                                self.service_check(t['name'] + '.ok', self.TASK_STATUS[t['state']], tags=task_tags)
+                                for key_name, (metric_name, metric_func) in self.TASK_METRICS.iteritems():
+                                    metric_func(self, metric_name, t['resources'][key_name], tags=task_tags)
+
+        stats_metrics = self._get_stats(url, timeout)
+        if stats_metrics:
+            tags = tags if tags else instance_tags
+            metrics = [self.SLAVE_TASKS_METRICS, self.SYSTEM_METRICS, self.SLAVE_RESOURCE_METRICS,
+                      self.SLAVE_EXECUTORS_METRICS, self.STATS_METRICS]
+            for m in metrics:
+                for key_name, (metric_name, metric_func) in m.iteritems():
+                    metric_func(self, metric_name, stats_metrics[key_name], tags=tags)
+
+        self.SERVICE_CHECK_NEEDED = True
```

Comment:
nitpick, but usually upper case variables are for constants which is not the case here.